### PR TITLE
modularize updater code

### PR DESCRIPTION
**Pull request by zeke**
I refactored this code with an eye toward turning it into a module that folks can `npm install` and integrate easily into their apps:

```js
require('whatever-we-call-this-updater-thing')({
  repo: 'ummoji/ummoji-desktop'
})
```

Options:

- `repo` String (required) - A GitHub repository in the format `owner/repo`
- `host` String (optional) - Defaults to `https://electron-update-server.herokuapp.com`
- `updateInterval` Integer (optional) - How frequently to check for updates in milliseconds. Defaults to `60000` (one minute)
- `debug` Boolean (optional) - Display debug output. Defaults to `false`

cc @juliangruber @marshallofsound

Haven't tested this. Would love to hear your feedback on the general idea.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
 require('electron-analytics').init('S1guTYsxsl')
 const {app} = require('electron')
-const updater = require('./app/lib/updater')
+
+require('./app/lib/updater')({
+  repo: 'ummoji/ummoji-desktop'
```
Comment (juliangruber) [inline main.js]: can we try to read this from the package.json?

File: app/lib/updater.js [hunk 2]
```diff
@@ -1,19 +1,58 @@
...
+    initUpdater(opts)
+  } else {
+    app.on('ready', () => initUpdater(opts))
+  }
+}
+
+module.exports = function initUpdater (opts) {
+  const defaults = {
+    host: 'https://electron-update-server.herokuapp.com',
+    updateInterval: 60 * 1000,
```
Comment (juliangruber) [inline app/lib/updater.js]: we might let people pass strings as well, which could be parsed by the `ms` module?

**Review by juliangruber — COMMENTED** [review]
otherwise LGTM!

File: app/lib/updater.js (hunk 2 above)
Comment (zeke) [reply app/lib/updater.js]: 👍

File: main.js (hunk 1 above)
Comment (zeke) [reply main.js]: Some apps use a two-package.json configuration popularized by `electron-builder`. In this scenario, the top-level package.json (which probably contains the `repository` field) will not be included in the bundled app. We had a big headache with this when we tried to be clever in `electron-packager`. So experience is telling me to make this one piece of information explicit.

**Pull request merged**